ckan/new_authz.py

```diff
@@ -431,12 +431,13 @@ def auth_is_anon_user(context):
 
         See ckan/lib/base.py:232 for pylons context object logic
     '''
```

Comment:
I think it makes sense to rely on the user in the context as this is what the function is passed as param and the controllers are doing the `context = {'user': c.user or c.author}` thing all over the place.

Part of me is slightly anxious about someone having a custom validator on user names that allows dots and this causing problems, but I'm probably just being too much paranoid :)

@wardi I'm curious as to why this was needed when moving tests around (again, I'm happy with it)